src/common/BackTrace.cc

```diff
@@ -70,4 +74,66 @@ void BackTrace::print(std::ostream& out) const
   }
 }
 
+void BackTrace::dump(Formatter *f) const
+{
+  f->open_array_section("backtrace");
+  for (size_t i = skip; i < size; i++) {
+    //      out << " " << (i-skip+1) << ": " << strings[i] << std::endl;
+
+    size_t sz = 1024; // just a guess, template names will go much wider
+    char *function = (char *)malloc(sz);
+    if (!function)
+      return;
+    char *begin = 0, *end = 0;
+
+    // find the parentheses and address offset surrounding the mangled name
+#ifdef __FreeBSD__
+    static constexpr char OPEN = '<';
+#else
+    static constexpr char OPEN = '(';
+#endif
+    for (char *j = strings[i]; *j; ++j) {
+      if (*j == OPEN)
+	begin = j+1;
+      else if (*j == '+')
+	end = j;
+    }
+    if (begin && end) {
+      int len = end - begin;
+      char *foo = (char *)malloc(len+1);
+      if (!foo) {
+	free(function);
+	return;
+      }
+      memcpy(foo, begin, len);
+      foo[len] = 0;
+
+      int status;
+      char *ret = nullptr;
+      // only demangle a C++ mangled name
+      if (foo[0] == '_' && foo[1] == 'Z')
```

Comment:
I don't think the test is necessary, and it makes me worry about portability; the result should be the same if it's not a mangled name, right?

Reply:
Still believe this is overcomplex

Reply:
could be, sorry.. i didn't look closely since this was literally a copy/paste of the non-formatted method.  we can probably clean them both up as a follow-on